packages/apollo-server/src/express.ts

```diff
@@ -20,16 +21,17 @@ export class ApolloServer extends ApolloServerBase<
     const { app, request } = config;
 
     app.use(config.path, cors(config.cors), json(), (req, res, next) => {
+      // only show graphiql if we want html for a GET
+      const accept = accepts(req);
```

Comment:
Since we only need to run the `Accept` header checking in the event that the other conditions on line 34 are met (`config.graphiql !== false && req.method === 'GET'`), we should defer this additional checking until we've confirmed we've needed it.

By short-circuiting, we'll avoid the more expensive logic tests, and this _the_ main entry-point.  i.e.:

```js
if (config.graphiql !== false && req.method === 'GET') {
  // Then check `Accept`, and return GUI if it wants HTML.
}
```